tests/hydra.rs

```diff
@@ -460,62 +462,107 @@ fn scenario_2() -> TestResult {
 }
 
 #[test]
-fn hydra_oura_stdout_scenario_1() -> TestResult {
```

Comment:
IMO we don't need `TestResult` — we can just as well use `.unwrap()`

Reply:
If I understand well TestResult as we have guarrantee that all errors from the body are going to caught and presented. No?

Reply:
I compared the output below. We get proper test failure and reporting both with `unwrap` and with `TestResult`. `TestResult`'s output does appear a bit cleaner, but it also doesn't include details about the source location where the error is thrown, which could be important if there are multiple potential sources in a single test.

With `TestResult`:

```shell
cargo test hydra                                                                                                           (1m 35s 341ms)
   Compiling oura v2.0.0-alpha.4 (/Users/anviking/IOHK/oura)
    Finished `test` profile [unoptimized + debuginfo] target(s) in 2.37s
     Running unittests src/lib.rs (target/debug/deps/oura-67d7c0d22115f1a7)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 13 filtered out; finished in 0.00s

     Running unittests src/bin/oura/main.rs (target/debug/deps/oura-b6d6ddaf4d6378cf)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/hydra.rs (target/debug/deps/hydra-43a1c17caf70064c)

running 4 tests
test hydra_restore_from_intersection_failure ... FAILED
test hydra_restore_from_intersection_success ... ok
test hydra_oura_stdout_scenario_2 ... ok
test hydra_oura_stdout_scenario_1 ... ok

failures:

---- hydra_restore_from_intersection_failure stdout ----
Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }


failures:
    hydra_restore_from_intersection_failure

test result: FAILED. 3 passed; 1 failed; 0 ignored; 0 measured; 9 filtered out; finished in 3.03s
```


With `unwrap`:
```shell
cargo test hydra                                                                                                               (33s 65ms)
   Compiling oura v2.0.0-alpha.4 (/Users/anviking/IOHK/oura)
    Finished `test` profile [unoptimized + debuginfo] target(s) in 1.71s
     Running unittests src/lib.rs (target/debug/deps/oura-67d7c0d22115f1a7)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 13 filtered out; finished in 0.00s

     Running unittests src/bin/oura/main.rs (target/debug/deps/oura-b6d6ddaf4d6378cf)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/hydra.rs (target/debug/deps/hydra-43a1c17caf70064c)

running 4 tests
test hydra_restore_from_intersection_failure ... FAILED
test hydra_restore_from_intersection_success ... ok
test hydra_oura_stdout_scenario_1 ... ok
test hydra_oura_stdout_scenario_2 ... ok

failures:

---- hydra_restore_from_intersection_failure stdout ----
thread 'hydra_restore_from_intersection_failure' panicked at tests/hydra.rs:493:54:
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    hydra_restore_from_intersection_failure

test result: FAILED. 3 passed; 1 failed; 0 ignored; 0 measured; 9 filtered out; finished in 3.03s

error: test failed, to rerun pass `--test hydra`
```


With `unwrap` and `RUST_BACKTRACE=1`:
```shell
 ǃ  ~/I/oura  ╍╍ anviking/better-intersection +  RUST_BACKTRACE=1 cargo test hydra                                                                                        (101) (6s 361ms)
    Finished `test` profile [unoptimized + debuginfo] target(s) in 0.25s
     Running unittests src/lib.rs (target/debug/deps/oura-67d7c0d22115f1a7)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 13 filtered out; finished in 0.00s

     Running unittests src/bin/oura/main.rs (target/debug/deps/oura-b6d6ddaf4d6378cf)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/hydra.rs (target/debug/deps/hydra-43a1c17caf70064c)

running 4 tests
test hydra_restore_from_intersection_failure ... FAILED
test hydra_restore_from_intersection_success ... ok
test hydra_oura_stdout_scenario_1 ... ok
test hydra_oura_stdout_scenario_2 ... ok

failures:

---- hydra_restore_from_intersection_failure stdout ----
thread 'hydra_restore_from_intersection_failure' panicked at tests/hydra.rs:493:54:
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }
stack backtrace:
   0: rust_begin_unwind
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panicking.rs:74:14
   2: core::result::unwrap_failed
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/result.rs:1679:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/result.rs:1102:23
   4: hydra::scenario_4
             at ./tests/hydra.rs:493:5
   5: hydra::hydra_restore_from_intersection_failure
             at ./tests/hydra.rs:497:19
   6: hydra::hydra_restore_from_intersection_failure::{{closure}}
             at ./tests/hydra.rs:496:45
   7: core::ops::function::FnOnce::call_once
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
   8: core::ops::function::FnOnce::call_once
             at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    hydra_restore_from_intersection_failure

test result: FAILED. 3 passed; 1 failed; 0 ignored; 0 measured; 9 filtered out; finished in 3.04s

error: test failed, to rerun pass `--test hydra`
```

Reply:
ok. thanks for the comparison :+1: